### DISCUSSION TITLE
Nova Repeater field doesn't accept Fluent

### DISCUSSION THREAD

**Discussion by JesperFiltenborg** [laravel/nova-issues#5921] [Ideas & Feature Requests]
- Laravel Version: 8.47
- Nova Version: 4.27.12
- PHP Version: 8.1.13
- Database Driver & Version:  mariadb:10.4
- Operating System and Version: MAC OS 13.2.1
- Browser type and version: Microsoft Edge
- Reproduction Repository: https://github.com/###/###

### Description:
I can't use the new Repeater field with the Fluent class, due to it requiring a type of eloquent model.

Error message:
Laravel\\Nova\\Fields\\Repeater\\Presets\\JSON::get(): Argument #2 ($model) must be of type Illuminate\\Database\\Eloquent\\Model, Laravel\\Nova\\Support\\Fluent given, called in /var/www/html/vendor/laravel/nova/src/Fields/Repeater.php

### Detailed steps to reproduce the issue on a fresh Nova installation:
I am using the https://github.com/outl1ne/nova-settings package due to the customer needing to set settings on the fly that can change often on the project globally.
Install the package.
Go to nova service provider
Go to the nova settings function
Add the following:
` 
NovaSettings::addSettingsFields([
            Repeater::make('List of test variables')
                ->repeatables([
                    Text::make('Test Variable', 'test_var'),
                ]),
        ], [], 'Some Setting', 'some_setting');
`

**Comment by ghost** [comment]
Fixing this issue would also allow for using repeater field in the action form? If yes, that would be really, really nice!

**Comment by finquz** [comment]
Is the problem solved? I have the same problem. @salvisb what solution did you make?

**Reply by ghost** [reply]
Currently we are just looping through a collection and rendering regular fields. But that only solves one of several use cases...

Creating a field based on a relation:
```php
// MaterialGeneratorPropertyField
public static function make(MaterialGroup $materialGroup): array
{
    return $materialGroup->properties->map(function (Property $property) {
        return Select::make($property->name, $property->uuid)
            ->stacked()
            ->fullWidth()
            ->options(function () use ($property) {
                // fetch options...
            })
            ->searchable();
    })->toArray();
}
```
Registering the field in an action:
```php
public function fields(NovaRequest $request): array
{
    return [
        // other fields...
        ...MaterialGeneratorPropertyField::make($materialGroup),
    ];
}
```

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
